torch/csrc/distributed/rpc/tensorpipe_agent.h

```diff
@@ -11,12 +11,21 @@
 #include <c10d/Store.hpp>
 #include <torch/csrc/distributed/rpc/rpc_agent.h>
 
+#include <tensorpipe/tensorpipe.h>
+
 // Forward-declare the TensorPipe classes we need, to avoid including its
 // headers in PyTorch's ones and thus have it become a public dependency.
 
 namespace tensorpipe {
 
+#define TENSORPIPE_USE_CUDA_IPC_CHANNEL TENSORPIPE_HAS_CUDA_IPC_CHANNEL && defined(USE_CUDA)
+
 class CpuBuffer;
+
+#if TENSORPIPE_USE_CUDA_IPC_CHANNEL
+class CudaBuffer;
+#endif
```

Comment:
Once TensorPipe will use multiple CUDA channels (which is actually already the case, there's cuda_xth too now) we'll need CudaBuffer to be forward-declared if any of them is present. Or, in fact, more generally, we should declare it whenever TENSORPIPE_USE_CUDA is enabled, and that flag should be equal to PyTorch's USE_CUDA.
```suggestion
#if USE_CUDA
class CudaBuffer;
#endif
```